### PR TITLE
Added load_from_capsule functionality to TTRT and Python Bindings

### PR DESCRIPTION
**Pull request by vprajapati-tt**
- Added functionality with the magic of `py::capsule`
- Can now run the entire transformation from TTIR -> Flatbuffer -> Perf Modelling with only python calls.
- Fixed a small bug

### CODE REVIEW

**Review by nsmithtt — APPROVED** [review]
Capsule changes lgtm, @tapspatel to sign off on ttrt changes

File: runtime/tools/python/ttrt/common/api.py [hunk 1]
```diff
@@ -1089,49 +1089,78 @@ def check_constraints(self):
...
-            )
-
-            self.logging.debug(f"ttnn_binary_paths={ttnn_binary_paths}")
-            self.logging.debug(f"ttmetal_binary_paths={ttmetal_binary_paths}")
-
-            for path in ttnn_binary_paths:
-                bin = Binary(self.logger, self.file_manager, path)
+            if "binary" not in self:
+                # Load from Capsule instead. only TTNN Path is supported for now
+                bin = Binary(self.logger, self.file_manager, "", self["capsule"])
```
Comment (tapspatel) [inline runtime/tools/python/ttrt/common/api.py:1094]: where are you assigning this self["capsule"] argument?

Comment (vprajapati-tt) [reply runtime/tools/python/ttrt/common/api.py:1094]: Capsule is defined from the arguments passed into the `API.Perf` call.

ex:
```
c_args = API.Perf.registered_args
c_args['capsule'] = ttmlir.passes.ttnn_to_flatbuffer(module)

<...>
```

Comment (tapspatel) [reply runtime/tools/python/ttrt/common/api.py:1094]: gotcha, looks good!

**Comment by tapspatel** [comment]
lgtm! If you want, also write an optional section in the ttrt docs on this capsule example.

**Review by tapspatel — APPROVED** [review]

**Pull request merged**